pkg/storage/db.go

```diff
@@ -1107,6 +1117,62 @@ func (db *DB) releasePublicIP(ctx context.Context, tx *sql.Tx, arnID string, ip
 	return err
 }
 
+func (db *DB) assignResourceRelationship(ctx context.Context, tx *sql.Tx, arnID string, resource string, when time.Time) error {
+	const assignResourceRelationshipQueryUpdate = `
+update aws_resource_relationship
+set not_before = $1
+where related_arn_id = $2
+  and not_before = to_timestamp(0)
+  and not_after > $1
+  and arn_id = $3;`
+
+	const assignResourceRelationshipQueryInsert = `
+insert into aws_resource_relationship
+    (not_before, related_arn_id, arn_id)
+values ($1, $2, $3) on conflict do nothing ;`
+
+	res, err := tx.ExecContext(ctx, assignResourceRelationshipQueryUpdate, when, resource, arnID)
+	if err != nil {
+		return err
+	}
+	changedRows, err := res.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if changedRows != 0 {
+		return nil
+	}
+	_, err = tx.ExecContext(ctx, assignResourceRelationshipQueryInsert, when, resource, arnID)
+	return err
+}
+
+func (db *DB) releaseResourceRelationship(ctx context.Context, tx *sql.Tx, arnID string, resource string, when time.Time) error {
+	const releaseResourceRelationshipQueryUpdate = `
+update aws_resource_relationship
+set not_after=$1
+where related_arn_id = $2
+  and arn_id = $3;`
```

Comment:
I don't know how often this happens (if at all) but if we were to have to two records with the same `arn_id` and `related_arn_id` then this update statement would update both, so to be safe we might want to add `and not_after is null` to the where clause section. We have something similar here (line 1098) https://github.com/asecurityteam/asset-inventory-api/pull/77/files#diff-106eb1177cdcbe545695a4fdec2e2a32R1098

Reply:
Great point @dscgarcia i doubt we will hit this scenario but completely agree addition of `and not_after is null` is good to be safe.